Takes Properties/myrrc_Toggle take two mono to stereo play (doubletrack mode).lua

```diff
@@ -0,0 +1,24 @@
+-- @description Toggle take two mono to stereo play (doubletrack mode)
+-- @author myrrc
+-- @version 1.0
+-- @about For media item with two takes, toggle between playing one take a time and both takes playing panned100% left and right (doubletrack mode, useful for guitars).
+
+
+function msg(...) reaper.ShowConsoleMsg(string.format("%s\n", string.format(...))) end
+
+reaper.PreventUIRefresh(1)
+reaper.Undo_BeginBlock()
+
+if reaper.CountSelectedMediaItems(0) ~=1 then return msg("Only one media item must be selected") end
+item = reaper.GetSelectedMediaItem(0, 0)
+
+if reaper.GetMediaItemNumTakes(item) ~= 2 then return msg("Media item must contain two takes") end
+left, right = reaper.GetMediaItemTake(item, 0), reaper.GetMediaItemTake(item, 1)
+
+reaper.SetMediaItemInfo_Value(item, "B_ALLTAKESPLAY", (reaper.GetMediaItemInfo_Value(item, "B_ALLTAKESPLAY") + 1) % 2)
+reaper.SetMediaItemTakeInfo_Value(left, "D_PAN", (reaper.GetMediaItemTakeInfo_Value(left, "D_PAN") + 1) % -2)
+reaper.SetMediaItemTakeInfo_Value(right, "D_PAN", (reaper.GetMediaItemTakeInfo_Value(right, "D_PAN") + 1) % 2)
```

Comment:
Nice script! Can merge as-is, or some suggestions:

- It can set pan values beyond 100% if the initial value is not exactly 100% or 0%.

  ```suggestion
  reaper.SetMediaItemInfo_Value(item, "B_ALLTAKESPLAY", reaper.GetMediaItemInfo_Value(item, "B_ALLTAKESPLAY") ~ 1)
  reaper.SetMediaItemTakeInfo_Value(left, "D_PAN", math.floor(reaper.GetMediaItemTakeInfo_Value(left, "D_PAN") + 1) % -2)
  reaper.SetMediaItemTakeInfo_Value(right, "D_PAN", math.floor(reaper.GetMediaItemTakeInfo_Value(right, "D_PAN") + 1) % 2)
  ```

- Because each take are toggled individually, it doesn't handle well a starting condition where one of the takes is already hard-panned.

Reply:
The ~1 trick is cool!

Reply:
> it doesn't handle well a starting condition where one of the takes is already hard-panned.
Let's leave this out for simplicity if you don't mind